dask/dataframe/tests/test_numeric.py

```diff
@@ -15,7 +15,7 @@ def test_to_numeric_on_scalars(arg):
 
 
 def test_to_numeric_on_dask_array():
-    arg = from_array(["1.0", "2", -3, 5.1])
+    arg = from_array(["1.0", "2", "-3", "5.1"])
```

Comment:
Hmm I'm curious why we need to make these all strings now

Reply:
The mixture of strings and floats seemed to cause issues in latest numpy:

```ipython
In [1]: import numpy as np

In [2]: np.asarray(["1.0", "2", -3, 5.1])
/home/julia/conda/envs/dask-upstream/lib/python3.8/site-packages/numpy/core/_asarray.py:102: FutureWarning: Promotion of numbers and bools to strings is deprecated. In the future, code such as `np.concatenate((['string'], [0]))` will raise an error, while `np.asarray(['string', 0])` will return an array with `dtype=object`.  To avoid the warning while retaining a string result use `dtype='U'` (or 'S').  To get an array of Python objects use `dtype=object`. (Warning added in NumPy 1.21)
  return array(a, dtype, copy=False, order=order)
Out[2]: array(['1.0', '2', '-3', '5.1'], dtype='<U32')
```

Reply:
the other option is to set the `dtype` explicitly I suppose.

Reply:
Interesting were they being coerced to `str`s implicitly before? If so, that might explain it

Do we know what use case this test was trying to cover? It seems odd to include `str`s and `float`s in an array with a non-struct `dtype`